packages/component-library/draft/Kaizen/Modal/presets/ConfirmationModal.tsx

```diff
@@ -31,6 +33,17 @@ interface Props {
 
 type ConfirmationModal = React.FunctionComponent<Props>
 
+const getIcon = type => {
+  switch (type) {
+    case "affirmative":
```

Comment:
where is `affirmative` coming from?

Reply:
@jhuangnz Looks like it's called `positive` in line 25.

Reply:
We could probably type the `type` parameter of this function with the same type as the `type` prop, and TypeScript would catch this sort of thing for us.

Reply:
Or if `getIcon` were made a regular function instead of an anonymous arrow function, I think TypeScript might be smart enough to automatically infer the type of the `type` argument. This works in Flow but not sure about TypeScript :)

Reply:
😅 good question! Wires crossed today... thanks!

Reply:
@lijuenc do you mean something like:
```
const getIcon = (type: Props["type"]) => {
  ...
}
```

Reply:
Close, but that looks like invalid syntax. Something like:

```
type ModalType = "positive" | "informative" | "negative" | "cautionary"

interface Props {
  ...
  type: ModalType
  ...
}

const getIcon = (type: ModalType) => {
  ...
}
```

Reply:
Thanks @lijuenc. Ha! My editor actually made the suggestion! But I'm not sure if it would have worked. I have amended with the syntax you have suggested.